core/src/main/java/org/apache/accumulo/core/conf/Property.java

```diff
@@ -230,6 +230,8 @@
   // properties that are specific to master server behavior
   MASTER_PREFIX("master.", null, PropertyType.PREFIX,
       "Properties in this category affect the behavior of the master server"),
+  MASTER_HOST("master.bind.address", "0.0.0.0", PropertyType.HOSTLIST,
```

Comment:
Regarding this naming convention, should these be `*.bind.address` or `*.address.bind`? What do you think? `master.bind.address` reads more naturally, but `master.address.bind` supports logically grouping address-related configs in a configuration hierarchy, which can be better for parsing, and more easily incorporate future changes.

An example of a future change which could benefit from the `*.address.bind` form could include work that address an explicit public advertisement address, in the case of the bind address not being publicly reachable (in the case of more complicated networking setups, such as those on some cloud services' infrastructure):

Example:
```ini
; to add a publicly reachable service address (the one recorded in ZooKeeper for clients trying
; to reach this server), in addition to the bind address we would need a public address:
master.bind.address = 172.16.0.1
master.public.address = 203.0.113.1

; alternative naming convention
master.address.bind = 0.0.0.0
master.address.public = 203.0.113.1 # or accumulo-master.example.com b/c this can be a hostname

; another possible future change could include ports to replace `*.port.client`,
; so we reduce the number of config options, and put address information all in one place
master.address.bind = 0.0.0.0:9999
master.address.public = accumulo-master.example.com:50999
```

I'm leaning towards the `<service>.address.<addressType>` naming convention, rather than the `<service>.<addressType>.address` convention that you currently have here, but am open to discussion... because naming is hard. What do you think?

Reply:
Naming is always hard.  Your suggestion to make it *.address.bind works for me.  I'm surprised there were no comments on the Property name MASTER_HOST - I like short if I can get away with it but maybe it should be MASTER_ADDRESS or even MASTER_BIND_ADDRESS

Reply:
I'm not concerned about the enum name at all, since that's not user-facing. It's internal only.

At one point I thought about making the property names derived from the enum names (basically, make lowercase, and replace underscores with dots, all in code). I did this at one point for some enums in our MapReduce code. But, it would require us to get pretty strict about enum naming in Property.java, and at this point, I don't think it's worth it.

Long story short: the enum can be whatever. :smiley_cat: